### PR TITLE
Finish declaring optional dependencies; fix crash when having no awards

### PR DESCRIPTION
**Pull request by Wuzzy2**
This PR (hopefully) finally makes sure that all award dependencies are now truly optional and properly declared.
Many awards depend on `default` and they are now only added when the `default` mod is present. I know, the first commit looks scary but all I did was to move the awards around and fixed indentation for the conditional I added. There were no changes to the actual award contents.
This means that all current awards now have a dependency on some mod.

This implies that it is possible for this mod to have 0 awards initially when it is used in an entirely different subgame. This special case is catched by displaying error messages instead of crashing.

These changes will allow completely new subgames to start with a completely empty set of awards to add their own, while still adding our own awards into Minetest Game.

### CODE REVIEW

**Comment by rubenwardy** [comment]
:+1: looks good

File: api.lua [hunk 1]
```diff
@@ -454,7 +467,7 @@ function awards.show_to(name, to, sid, text)
 		end
 		-- Show formspec to user
 		minetest.show_formspec(to,"awards:awards",
-			"size[11,5]" .. awards.getFormspec(name, to, sid))
+			awards.getFormspec(name, to, sid))
```
Comment (rubenwardy) [inline api.lua]: this will cause problems with sfinv and maybe unified inventory

Comment (Wuzzy2) [inline api.lua]: I don't know what to do about it.

**Comment by rubenwardy** [comment]
a04f3df1e2996706c23d8562d2684a41a7215a05
ae01ea01dd80ae40a8289e57908586ec70bfa5b6

**Pull request closed without merging**